src/ipyradiant/visualization/explore/graph_explorer.py

```diff
@@ -0,0 +1,190 @@
+
+    _sparql = """
+        SELECT DISTINCT ?s ?label
+        WHERE {{
+            ?s a ?type .
+            OPTIONAL {{ ?s rdfs:label ?label . }}
+
+            VALUES ({}) {{
+                {}
+            }}
+        }}
+    """
+    values = None
+
+    @property
+    def sparql(cls):
+        return build_values(cls._sparql, cls.values)
+
+
+class RDFTypeSelectMultiple(W.VBox):
+    """TODO improve docs, and rename attrs and methods?
+    Used to build a selection for available types in the graph.
+    Uses a library query. Uses graph object's namespace.
+    """
+
+    graph = T.Instance(Graph, allow_none=True)
+    label = T.Instance(W.HTML)
+    select_widget = T.Instance(SelectMultipleURI)
+
+    def __init__(self, label: str = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        label = label or "Available types:"
+        self.label.value = f"<b>{label}</b>"
+        self.children = tuple([self.label, self.select_widget])
+
+    def make_available_types(self):
+        if self.graph is not None:
+            ns = dict(self.graph.namespaces())
+            types = AllTypes.run_query(self.graph)
+            # TODO how to allow user specification of custom CustomURI class
+            self.select_widget.pithy_uris = tuple(
+                map(lambda x: CustomURI(x, namespaces=ns), types["o"].to_list())
+            )
+
+    @T.default("label")
+    def make_default_label(self):
+        return W.HTML("<b>Available types:</b>")
+
+    @T.default("select_widget")
+    def make_default_select_widget(self):
+        return SelectMultipleURI(rows=10)
+
+    @T.observe("graph")
+    def update_graph(self, change):
+        self.make_available_types()
+
+
+class RDFSubjectSelectMultiple(W.VBox):
+    # Define the query class
+    class SubjectsOfType(SPARQLQueryFramer, metaclass=MetaSubjectsOfType):
+        values = None  # note: query will not run without values
+
+    graph = T.Instance(Graph, allow_none=True).tag(default=None)
+    query = SubjectsOfType
+    _values = T.Instance(dict, allow_none=True).tag(default=None)
+    #
+    label = T.Instance(W.HTML)
+    select_widget = T.Instance(SelectMultipleURI)
+
+    def __init__(self, label: str = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        label = label or "Available subjects:"
+        self.label.value = f"<b>{label}</b>"
+        self.children = tuple([self.label, self.select_widget])
+
+    @T.default("label")
+    def make_default_label(self):
+        # TODO update_label(label_str: str):
+        #  label should be a property that is based on label_text
+        return W.HTML("<b>Available subjects:</b>")
+
+    @T.default("select_widget")
+    def make_default_select_widget(self):
+        return SelectMultipleURI(rows=10)
+
+    def update_select(self):
+        if self.graph is not None and self._values is not None:
+            qres = self.run_query()
+            sw_uris = []
```

Comment:
List comprehension?

```python
sw_uris = [
    CustomItem(
        _repr=lambda x: f"{x.label}:   ->   {x.uri}",
        uri=uri,
        label=label,
    )
    for uri, label in qres.values
]
```

Reply:
maybe even remove the `qres`?

`for uri, label in self.run_query().values`

...since it is only used once and it is not really adding a lot of syntactic meaning...